src/openloco/windows/constructionwnd.cpp

```diff
@@ -1,75 +1,4836 @@
+        {
+            auto bridge = trackObj->bridges[i];
+            if (bridge == 0xFF)
+                continue;
+            auto bridgeObj = objectmgr::get<bridge_object>(bridge);
+            if (currentYear < bridgeObj->designed_year)
+                continue;
+            _bridgeList[bridgeCount] = bridge;
+            bridgeCount++;
+        }
+
+        for (uint8_t i = 0; i < objectmgr::get_max_objects(object_type::bridge); i++)
+        {
+            auto bridgeObj = objectmgr::get<bridge_object>(i);
+            if (bridgeObj == nullptr)
+                continue;
+            for (auto j = 0; j < bridgeObj->var_22; j++)
+            {
+                if (trackType != bridgeObj->var_22)
+                    continue;
+                if (currentYear < bridgeObj->designed_year)
+                    continue;
+                for (size_t k = 0; k < std::size(_bridgeList); k++)
+                {
+                    auto bridge = bridgeListCopy[k];
+                    if (bridgeListCopy == _bridgeList)
+                    {
+                        _bridgeList[bridgeCount] = i;
+                        bridgeCount++;
+                        break;
+                    }
+                    if (i == bridge)
+                        break;
+                }
+            }
+        }
+
+        _bridgeList[bridgeCount] = 0xFF;
+
+        sortList(_bridgeList, std::size(_bridgeList));
+    }
+
+    // 0x004A693D
+    static void refreshTrackModList()
+    {
+        auto trackType = _trackType;
+        auto companyId = _updatingCompanyId;
+        _modList[0] = 0xFF;
+        _modList[1] = 0xFF;
+        _modList[2] = 0xFF;
+        _modList[3] = 0xFF;
+        auto flags = 0;
+
+        for (uint8_t vehicle = 0; vehicle < objectmgr::get_max_objects(object_type::vehicle); vehicle++)
+        {
+            auto vehicleObj = objectmgr::get<vehicle_object>(vehicle);
+
+            if (vehicleObj == nullptr)
+                continue;
+
+            if (vehicleObj->mode != TransportMode::rail)
+                continue;
+
+            if (trackType != vehicleObj->track_type)
+                continue;
+
+            auto edi = vehicle;
+            auto edx = vehicle & 0x1F;
+            edi = edi >> 5;
+
+            auto company = companymgr::get(companyId);
+
+            if (!(company->unlocked_vehicles[edi] & 1 << edx))
+                continue;
+
+            for (auto i = 0; i < vehicleObj->num_mods; i++)
+            {
+                flags |= 1ULL << vehicleObj->required_track_extras[i];
+            }
+
+            if (!(vehicleObj->flags & flags_E0::rack_rail))
+                continue;
+
+            flags |= 1ULL << vehicleObj->rack_rail_type;
+        }
+
+        auto trackObj = objectmgr::get<track_object>(trackType);
+
+        for (auto i = 0; i < trackObj->num_mods; i++)
+        {
+            if (flags & (1 << trackObj->mods[i]))
+                _modList[i] = trackObj->mods[i];
+        }
+    }
+
```

Comment:
The above functions are very similar between road and track can you look into merging them together as much as possible so that there is no requirement for the duplicated versions.